Documentation/check-links.sh

```diff
@@ -0,0 +1,47 @@
+#!/usr/bin/env bash
+set -o errexit
+set -o nounset
+set -o pipefail
+
+script_dir="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
+
+build_dir="${script_dir}/_build"
+warnings="${build_dir}/links_warnings.txt"
+filtered_warnings="${build_dir}/filtered_links_warnings.txt"
+linksdir="${build_dir}/links"
+
+cd "${script_dir}"
+mkdir -p "${build_dir}"
+rm -f -- "${warnings}"
+rm -f -- "${filtered_warnings}"
+
+filter_warnings() {
+    [ -s "${warnings}" ] || return
+    grep -v -E \
+        -e "/_api/v1/.*/README\.md:[0-9]+: WARNING: 'myst' reference target not found:" \
+        -e "/.*:[0-9]+: WARNING: circular inclusion in \"include\" directive" \
+        "${warnings}"
+}
+
+has_build_warnings() {
+    filter_warnings > /dev/null
+}
+
+build_with_linkchecker() {
+    rm -rf "${linksdir}"
+    
+    sphinx-build -b linkcheck -d "${build_dir}/doctrees" . "${linksdir}" \
```

Comment:
Nit: trailing spaces

```suggestion
    rm -rf "${linksdir}"

    sphinx-build -b linkcheck -d "${build_dir}/doctrees" . "${linksdir}" \
```